components/patient-data-sharing/receiver-api/src/main/java/org/phenotips/data/receive/internal/DefaultReceivePatientData.java

```diff
@@ -545,6 +545,8 @@ public JSONObject receivePatient()
                 this.logger.warn("Created new patient successfully");
             }
 
+            // get updated saved instance of patient document
+            affectedPatient = this.patientRepository.get(affectedPatient.getDocument());
```

Comment:
I think we only need to do this if we changed permissions in all other cases it would be the same object